### ISSUE TITLE
PRM, loss nan

### ISSUE THREAD

**Issue by EthanChen1234** [OpenRLHF/OpenRLHF#509]
感谢开源，非常干净的训练框架。

在PRM训练中，出现loss为None.

### 启动脚本
`export CUDA_VISIBLE_DEVICES=0

deepspeed --module openrlhf.cli.train_prm \
   --save_steps 500 \
   --logging_steps 1 \
   --eval_steps 100 \
   --train_batch_size 64 \
   --micro_train_batch_size 2 \
   --pretrain /data0/models/llama3.2_3b_it  \
   --bf16 \
   --max_epochs 1 \
   --max_len 8192 \
   --zero_stage 3 \
   --learning_rate 1e-6 \
   --dataset /data0/esun.chen/data/Math-Shepherd-zhuzilin/ \
   --input_key input \
   --label_key value \
   --flash_attn \
   --load_checkpoint \
   --gradient_checkpointing \
   --packing_samples \
   --wandb_group prm \
   --placeholder_token ки \
   --reward_tokens + -`

### 代码修改
模型，采用 llama3.2_3b_it
process_reward_dataset.py
`# self.placeholder_token_id = convert_token_to_id(self.placeholder_token, self.tokenizer)
self.placeholder_token_id = 116624  # 'ки' -> 171165, ' ки' -> 116624
`

`
Train epoch:   0%|                                                           | 0/1 [00:00<?, ?it/sStarting from v4.46, the `logits` model output will have the same type as the model (except at train time, where it will always be FP32)
/opt/conda/lib/python3.10/site-packages/torch/utils/checkpoint.py:1399: FutureWarning: `torch.cpu.amp.autocast(args...)` is deprecated. Please use `torch.amp.autocast('cpu', args...)` instead.
  with device_autocast_ctx, torch.cpu.amp.autocast(**cpu_autocast_kwargs), recompute_context:  # type: ignore[attr-defined]

Train step of epoch 0:   0%| | 141/222327 [01:53<47:40:44,  1.29it/s, **prm_loss=nan, loss_mean=nan**,
`

**Comment by hijkzzz** [comment]
llama3 不要用 ки 
请找一个 reserved_token 代替 ки

**Comment by EthanChen1234** [comment]
@hijkzzz  好的，可以正常训练了。

非常感谢及时回复。

**Comment by WeepCat** [comment]
> llama3 不要用 ки 请找一个 reserved_token 代替 ки

您好，我还有一个问题，指定 reward_tokens 是 "+" 和 "-"，但是 tokenizer 切词完之后的目标 label 和 placeholder_token 位置可能不一致，我看代码中似乎没有考虑这一点？
